int32 do_trajectory_task
